src/platform/site-wide/user-nav/containers/Main.jsx

```diff
@@ -58,10 +54,35 @@ export class Main extends React.Component {
   }
 
   setToken = (event) => {
-    if (event.data === sessionStorage.userToken) { this.props.initializeProfile(); }
+    if (event.data === sessionStorage.userToken) {
+      this.executeRedirect();
+      this.props.initializeProfile();
+    }
   }
```

Comment:
Here, `executeRedirect` is called again, this time immediately after receiving the token from the sign-in window and bypassing the profile-initialization logic if a redirect is performed.